### PR TITLE
Use build stage refactor travis config file and add clang tsan test

### PR DESCRIPTION
**Pull request by zyfjeff**
### Describe what this PR does / why we need it
Use clang tsan to detect data race problems in my code

### Does this pull request fix one issue?

N/A

### Describe how you did it

1. `bazel build -c dbg --config=clang-tsan //tests/... `
2. `./bazel-bin/tests/tsan-flow`

### Describe how to verify it

Is the exit code of `./bazel-bin/tests/tsan-flow` 0?

### CODE REVIEW

**Comment by zyfjeff** [comment]
The current code has a data race problem so tsan's test will fail, which is normal.

File: tests/tsan-flow.cc [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+#include "sentinel-core/init/init_target_registry.h"
+#include "sentinel-core/log/log_base.h"
+#include "sentinel-core/log/logger.h"
+#include "sentinel-core/log/metric/metric_log_task.h"
+#include "sentinel-core/public/sph_u.h"
+#include "sentinel-core/transport/command/http_server_init_target.h"
+
+std::atomic<int> g_count{0};
+
+Sentinel::Init::InitTargetRegister<
```
Comment (sczyh30) [inline tests/tsan-flow.cc]: Duplicate initialization of `HttpCommandCenterInitTarget`?

Comment (zyfjeff) [reply tests/tsan-flow.cc]: Sorry, it is indeed redundant, I will delete it.

**Comment by sczyh30** [comment]
Thanks!

**Pull request merged**